File: LeetCode/1446. Consecutive Characters.js
// https://leetcode.com/problems/consecutive-characters/description/

const { test } = require('../../test');

const maxPower = function (str) {
  let currentCount = 0;

  let max = 0;

  for (let i = 0; i < str.length; i++) {
    if (str[i] === str[i - 1]) {
      currentCount++;
    } else {
      currentCount = 1;
    }

    max = Math.max(currentCount, max);
  }

  return max;
};

test(maxPower, [
  {
    input: ['leetcode'],
    expected: 2,
  },
  {
    input: ['abbcccddddeeeeedcba'],
    expected: 5,
  },
]);

File: LeetCode/1446. Consecutive Characters.py
# https://leetcode.com/problems/consecutive-characters/description/

from test import test


class Solution:
    def maxPower(self, string: str) -> int:
        counter = 0
        power = 0

        index = 0

        str_length = len(string)

        while index < str_length:
            if index == 0 or string[index] != string[index - 1]:
                counter = 1
            else:
                counter += 1

            power = max(power, counter)

            index += 1

        return power


solution = Solution()


test(
    solution.maxPower,
    [
        {
            "input": ["leetcode"],
            "expected": 2,
        },
        {
            "input": ["abbcccddddeeeeedcba"],
            "expected": 5,
        },
    ],
)
